pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2017-2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/api/go/profile"
+	"github.com/lf-edge/eve/pkg/pillar/flextimer"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+const (
+	defaultLocalProfileServerPort = "8888"
+	savedLocalProfileFile         = "lastlocalprofile"
+)
+
+func getLocalProfileURL(localProfileServer string) (string, error) {
+	localProfileURL := fmt.Sprintf("http://%s", localProfileServer)
+	u, err := url.Parse(localProfileURL)
+	if err != nil {
+		return "", fmt.Errorf("url.Parse: %s", err)
+	}
+	if u.Port() == "" {
+		localProfileURL = fmt.Sprintf("%s:%s", localProfileURL, defaultLocalProfileServerPort)
+	}
+	return fmt.Sprintf("%s/api/v1/local_profile", localProfileURL), nil
+}
+
+// Run a periodic fetch of the currentProfile
+func localProfileTimerTask(handleChannel chan interface{},
+	getconfigCtx *getconfigContext) {
+
+	ctx := getconfigCtx.zedagentCtx
+	iteration := 0
+	localProfileServer := getconfigCtx.localProfileServer
+	if localProfileServer != "" {
+		localProfileURL, err := getLocalProfileURL(localProfileServer)
+		if err != nil {
+			log.Errorf("localProfileTimerTask: getLocalProfileURL: %s", err)
+		} else {
+			if err := getLocalProfileConfig(localProfileURL, iteration, getconfigCtx); err != nil {
+				log.Errorf("localProfileTimerTask: %s", err)
+			}
+		}
+		publishZedAgentStatus(getconfigCtx)
+	}
```

Comment:
if the localProfileServer is cleared by the controller, then EVE should forget about any profile received by the local server (including forget about the checkpoint) and use the profile from the controller. 
And that switch should be logged.

Reply:
Added `cleanSavedProtoMessage`

Reply:
I didn't mean to imply it should be done here.
But there has to be a clearly identified place where you handle changes to localProfileServer, so you can both log.Noticef them, and do the cleanup associated with localProfileServer being cleared by the controller.